### PR TITLE
feat: add `Intl` in es6 globals

### PR DESCRIPTION
**Pull request by aladdin-add**
see the discussion: https://github.com/sindresorhus/globals/pull/213#issue-2142294092

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x ] Other, please explain:

#### What changes did you make? (Give an overview)
In eslintrc, eslint was using the npm package `globals`. However, in the new config (aka flat config) eslint does not rely on `globals` but maintains its own es globals.

In the newly released globals v15. it moved the `Intl` from `globals.node`/`globals.browsers` to `globals.es2015`.

this commit added `Intl`, just like glboals v15 did.

### CODE REVIEW

**Comment by fasttime** [comment]
Good catch 👍🏻 I'll just note that this change has a potential for reporting more problems, for example with `no-redeclare` if users were using a global config comment to add the missing global ([repro](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlZGVjbGFyZTogXCJlcnJvclwiICovXG5cbi8qIGdsb2JhbCBNYXRoICovXG4vKiBnbG9iYWwgSW50bCAtLSB3aWxsIGJlIGFuIGVycm9yIHdoZW4gSW50bCBpcyBhIGJ1aWx0LWluIGdsb2JhbCAqL1xuIiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJzb3VyY2VUeXBlIjoibW9kdWxlIiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)).

```js
/* eslint no-redeclare: "error" */

/* global Intl -- will be an error when Intl is a built-in global */
```

So maybe it should be tagged as `feat:`?

**Comment by aladdin-add** [comment]
@fasttime thanks for the suggestion, I've updated the title.

**Comment by mdjermanovic** [comment]
Does either ECMA-262 or ECMA-402 mention that the existence of global `Intl` is mandatory for a conforming implementation of ECMAScript? I see that in https://github.com/sindresorhus/globals/pull/213 the argument for including `Intl` is the similarity to `JSON` in terms that they both have complementary specifications, but unlike `Intl`, ECMA-262 requires the existence of global `JSON`.

**Comment by aladdin-add** [comment]
All I know is that all modern js runtimes support it. would like to hear from @ljharb. :)

**Comment by ljharb** [comment]
Technically Intl is normative optional, so no, it’s not required. However, i don’t know of any runtime that’s feasible to use (omitting things like jsc, d8, etc) that omits it, although maybe Moddable does in xs?

**Comment by fisker** [comment]
If we only consider "required" objects, `escape` is not required either, it's an "annex b" feature.


BTW, `Object.prototype` properties like  `toString` removed from builtin globals in `globals@15` https://github.com/sindresorhus/globals/pull/237

**Comment by nzakas** [comment]
I don't feel strongly about adding vs. not, as it's pretty easy to declare a new global if the user needs it.

The larger issue for me is what users expect. My hunch is that they'd expect `Intl` to be predeclared in ESLint, so I'd be in favor of adding it somewhere even if not es6.

**Comment by aladdin-add** [comment]
we can add it in es6 globals; users can turn off (`"Intl": "off"`) in some rare cases. It seems a good default for me. wdyt? @mdjermanovic

**Comment by mdjermanovic** [comment]
Make sense to me. Seems that everyone agrees, so marking as accepted.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Since this is a `feat`, per our [policies](https://eslint.org/docs/latest/maintain/review-pull-requests#required-approvals-for-pull-requests) it requires one more approval before merging.

**Review by fasttime — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
